multidict/_multidict_py.py

```diff
@@ -120,6 +120,8 @@ def __eq__(self, other):
                 if i1 != i2 or v1 != v2:
                     return False
             return True
+        if len(self) != len(other):
```

Comment:
It must be possible to move this to before line `114` and remove check at line `117`. They are equal, so no need to have it copy-pasted in two places.

Reply:
I disagree.
Check from line 117 is faster than line 123: no python magic method `__len__` is involved.
Code duplication is very low, let's merge PR as is.

Reply:
If it's about speed, then I'd convert `len(self)` into `self._impl._items`